specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2023-12-10-preview/cloudhsm.json

```diff
@@ -0,0 +1,975 @@
+          "x-ms-enum": {
+            "name": "CloudHsmClusterSkuName",
+            "modelAsString": false
+          }
+        },
+        "capacity": {
+          "format": "int32",
+          "type": "integer",
+          "description": "Sku capacity"
+        }
+      }
+    },
+    "BackupProperties": {
+      "type": "object",
+      "description": "Cloud Hsm Cluster backup information",
+      "properties": {
+        "azureStorageBlobContainerUri": {
+          "type": "string",
+          "description": "Azure Blob storage container Uri"
+        },
+        "lastBackupDateTime": {
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true,
+          "description": "Last Date Time that Customer Enabled Backup was taken"
+        }
+      }
+    },
+    "RestoreProperties": {
+      "type": "object",
+      "description": "Cloud Hsm Cluster restore information",
+      "properties": {
+        "blobStoreDirectory": {
+          "type": "string",
+          "description": "Directory name in Azure Storage Blob where the backup is stored"
+        },
+        "azureStorageBlobContainerUri": {
+          "type": "string",
+          "description": "Azure Blob storage container Uri"
+        }
+      }
+    },
+    "CloudHsmClusterError": {
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/CHsmError"
+        }
+      },
+      "description": "The Cloud HSM Cluster error details",
+      "type": "object"
+    },
+    "CHsmError": {
+      "properties": {
+        "code": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The error code."
+        },
+        "message": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The error message."
+        }
+      },
+      "type": "object",
+      "description": "Error details"
+    },
+    "CloudHsmClusterListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CloudHsmCluster"
+          },
+          "description": "The list of Cloud HSM Clusters."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of Cloud HSM Clusters."
+        }
+      },
+      "description": "List of Cloud HSM Clusters",
+      "type": "object"
+    },
+    "PrivateEndpointConnection": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PrivateEndpointConnectionProperties",
+          "x-ms-client-flatten": true,
+          "description": "Resource properties."
+        },
+        "etag": {
+          "type": "string",
+          "description": "Modified whenever there is a change in the state of private endpoint connection."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
```

Comment:
>v3 [](http://example.com/codeflow?start=67&length=2)

Same here Why is this being changed to v3 from v5?

Reply:
There was a validation error and I was told that this might solve it. I will update back to v5.